runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1459,7 +1459,9 @@ private void streamingCommitLoop() {
     Commit commit = null;
     while (running.get()) {
       // Batch commits as long as there are more and we can fit them in the current request.
-      CommitWorkStream commitStream = streamPool.getStream();
+      // We lazily initialize the commit stream to make sure that we only create one after
+      // we have a commit.
+      CommitWorkStream commitStream = null;
```

Comment:
Why not move this declaration below to line 1492 instead of assigning null here.

Then you wouldn't have to check if things are null or not.

Reply:
We want to add a batch of things to the same stream and then call flush.
If the declaration was moved to 1492, the commit stream would only be used for a single call to CommitWorkItem.

It would be nice if this method could be broken into some helper methods.  However I don't see a clean way to do so because of the logic where a unbatched Commit needs to be consumed by a subsequent stream.

Reply:
Thanks, makes sense.